lib/spack/spack/operating_systems/cray_backend.py

```diff
@@ -86,9 +86,9 @@ def __init__(self):
             # distro.linux_distribution, while still calling __init__
             # methods further up the MRO, we skip LinuxDistro in the MRO and
             # call the OperatingSystem superclass __init__ method
-            super(LinuxDistro, self).__init__(name, version)
+            super().__init__(name, version)
```

Comment:
This breaks on Cray. The argument of `super()` is `LinuxDistro` and not the class itself `CrayBackend` (see code comment just above).

```
  File "/users/rmeli/spack-daint/lib/spack/spack/platforms/cray.py", line 84, in __init__
    back_distro = CrayBackend()
  File "/users/rmeli/spack-daint/lib/spack/spack/operating_systems/cray_backend.py", line 89, in __init__
    super().__init__(name, version)
TypeError: __init__() takes 1 positional argument but 3 were given
```

Reply:
Good catch! Fixed in #38905